### ISSUE TITLE
QNX build is broken

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmithdemos/leveldb#212]
```
When building for QNX (which has a strict C++ library), the build is broken. 
The problem is that a couple of headers are included using the "#include 
<c...>" style, but the symbols are used in the global (i.e. not std::) 
namespace.

Compiler errors are:

../../third_party/leveldatabase/src/include/leveldb/env.h:235:41: error: 
'va_list' has not been declared

and

../../third_party/leveldatabase/src/util/arena.h:21:9: error: expected ';' at 
end of member declaration
../../third_party/leveldatabase/src/util/arena.h:21:25: error: expected ')' 
before 'bytes'
../../third_party/leveldatabase/src/util/arena.h:24:9: error: expected ';' at 
end of member declaration
../../third_party/leveldatabase/src/util/arena.h:24:32: error: expected ')' 
before 'bytes'
../../third_party/leveldatabase/src/util/arena.h:29:3: error: 'size_t' does not 
name a type
../../third_party/leveldatabase/src/util/arena.h:34:9: error: expected ';' at 
end of member declaration
../../third_party/leveldatabase/src/util/arena.h:34:33: error: expected ')' 
before 'bytes'
../../third_party/leveldatabase/src/util/arena.h:35:9: error: expected ';' at 
end of member declaration
../../third_party/leveldatabase/src/util/arena.h:35:33: error: expected ')' 
before 'block_bytes'
../../third_party/leveldatabase/src/util/arena.h:39:3: error: 'size_t' does not 
name a type
../../third_party/leveldatabase/src/util/arena.h:45:3: error: 'size_t' does not 
name a type
../../third_party/leveldatabase/src/util/arena.h:52:30: error: 
'leveldb::Arena::Allocate' declared as an 'inline' variable
../../third_party/leveldatabase/src/util/arena.h:52:30: error: 'char* 
leveldb::Arena::Allocate' is not a static member of 'class leveldb::Arena'
../../third_party/leveldatabase/src/util/arena.h:52:30: error: 'size_t' was not 
declared in this scope
../../third_party/leveldatabase/src/util/arena.h:52:44: error: expected ',' or 
';' before '{' token
../../third_party/leveldatabase/src/util/arena.cc:68:1: error: expected '}' at 
end of input
```

---

Original issue reported on code.google.com by `efidler....@gmail.com` on 28 Oct 2013 at 10:17

**Comment by GoogleCodeExporter** [comment]
```
patch
```

---

Original comment by `efidler....@gmail.com` on 28 Oct 2013 at 10:19

Attachments:
- [0001-Fix-QNX-build-by-properly-importing-symbols-into-glo.patch](https://storage.googleapis.com/google-code-attachments/leveldb/issue-212/comment-1/0001-Fix-QNX-build-by-properly-importing-symbols-into-glo.patch)

**Comment by GoogleCodeExporter** [comment]
```
I'm not sure how to find a reviewer for this. I don't see any leveldb patches 
on codereview.appspot.com, so I'm not sure where they happen.
```

---

Original comment by `efidler....@gmail.com` on 11 Nov 2013 at 5:44

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 1.15
```

---

Original comment by `dgrogan@chromium.org` on 10 Dec 2013 at 7:49
- Changed state: **Fixed**
